docs/blueprints/EP036.rst

```diff
@@ -0,0 +1,237 @@
+:EP: 36
+:Title: Openflow multi-table pipeline processing
+:Authors:
+    - Aldo Ortega <aortegay AT fiu DOT edu>
+    - Vinicius Arcanjo <vindasil AT fiu DOT edu>
+    - Italo Valcy <idasilva AT fiu DOT edu>
+
+:Created: 2023-05-04
+:Updated: 2023-10-04
+:Kytos-Version: 2022.3
+:Status: Draft
+
+***************************************************
+EP36 - Openflow multi-table pipeline processing
+***************************************************
+
+Abstract
+========
+
+This blueprint will present the requirements for the implementation of switch multi-tables through new NApp called **of_multi_table**.
+
+Motivation
+==========
+
+Multi-table processing will optimize resources in the data plane pipeline implementation. Because we are implementing one pipeline structure to all switches, it is necessary to ensure the consistency of tables to avoid missing packets thus a NApp will take full responsability of this process.
+
+Specification
+=============
+
+Each Kytos-ng Napp that requests OpenFlow flows, must define which table will be used to install flows in a configuration attribute.
+Also, a new NApp **of_multi_table** that will initially have the following responsibilities:
+
+  - Know which table is enabled.
+  - Ensuring the existence of a reasonable table-miss flow
+  - Publish events to handle the assigned table of each NApp
+  - In charge of modifying and analyzing **pipelines** collection
+  - The cookie prefix is **0xAD** for its posted flows.
+  - Requesting flow deletions and installations on behalf of other NApps when a pipeline changes.
+
+Pipeline Status: Enabled, Disabled
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The status of a pipeline dictates wheter it is applied as the current multi table system in every switch. Once a pipeline is ``enabled``, the previous one will be automatically ``disabled`` ensuring that there is always 1 or 0 (default pipeline is not counted) pipelines enabled.
+
+Default Pipeline
+~~~~~~~~~~~~~~~~
+
+The default pipeline is every flow in the table 0.
+
+Table Group
+~~~~~~~~~~~
+
+**of_multi_table** will let know other NApp about their determined table. Since it is not neccessary to send the entire pipeline, the content of the event **kytos/multi_table.enable_table** is obtained from the pipeline key ``napps_table_groups`` value. A value example could be ``{"mef_eline": ["epl"]}`` and it means that **mef_eline** EPL flows should be installed in the table where this value came from.

```

Comment:
```suggestion
**of_multi_table** will let know other NApp about their determined table. Since it is not necessary to send the entire pipeline, the content of the event **kytos/multi_table.enable_table** is obtained from the pipeline key ``napps_table_groups`` value. A value example could be ``{"mef_eline": ["epl"]}`` and it means that **mef_eline** EPL flows should be installed in the table where this value came from.

```

tiny suggestion for a typo that my spell checker has caught.